transcom/mymove
# B-20671 Parse TPPS Paid Invoice INT Part 2

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20671)
## [INT PR Part 1](https://github.com/transcom/mymove/pull/13434)
## Summary

The first round of tests written for this PR used example test files that were downloaded from the SFTP pickup folder and opened in vscode. However, those files were found to have a slightly different encoding while still directly inside the pickup folder.

Downloaded/opened in vscode:
![image](https://github.com/user-attachments/assets/32696c7c-69aa-4113-aa1e-94105d0dddc8)

While still in the SFTP pickup folder directly:
![image](https://github.com/user-attachments/assets/6bee81cd-2367-4ab9-9df0-07519d223533)

This new test parses the version of the file that is directly inside of the SFTP directory. I also fixed a few bugs in the parsing logic, specifically where the getPriceParts was not properly handing non-decimal values (as it should in the case of even dollar amounts with no change).

After this test, we now have three different ways of testing the parsing logic:
1. Parsing a string representation of the file.
2. Parsing and storing in the database a version of the file that has been downloaded from the SFTP pickup folder and viewed in vscode.
3. Parsing and storing in the database a version of the file that is directly from the SFTP pickup folder, with slightly different encoding than the previous step.

### How to test

1. Open the new `pkg/services/invoice/fixtures/tpps_paid_invoice_report_testfile_tpps_pickup_dir.csv` file in vscode and take a look at how it's formatted.
2. In `pkg/services/invoice/process_tpps_paid_invoice_report_test.go`, verify that the `successfully proccesses a TPPSPaidInvoiceReport from a file directly from the TPPS pickup directory and stores it in the database` test passes, as well as the previous parsing tests.

## Screenshots

This shows the SFTP pickup folder and the files inside.
![image](https://github.com/user-attachments/assets/ff6ef673-8310-4ede-99a9-a1ca97884f73)
